numpyro/distributions/directional.py

```diff
@@ -153,6 +153,138 @@ def variance(self):
+    :class:`~numpyro.infer.NUTS`. For example, the following will produce a prior over
+    skewness for the 2-torus,::
+
+        @numpyro.handlers.reparam(config={'phi_loc': CircularReparam(), 'psi_loc': CircularReparam()})
+        def model(obs):
+            # Sine priors
+            phi_loc = numpyro.sample('phi_loc', VonMises(pi, 2.))
+            psi_loc = numpyro.sample('psi_loc', VonMises(-pi / 2, 2.))
+            phi_conc = numpyro.sample('phi_conc', Beta(1., 1.))
+            psi_conc = numpyro.sample('psi_conc', Beta(1., 1.))
+            corr_scale = numpyro.sample('corr_scale', Beta(2., 5.))
+
+            # Skewing prior
+            ball_trans = L1BallTransform()
+            skewness = numpyro.sample('skew_phi', Normal(0, 0.5).expand((2,)))
+            skewness = ball_trans(skewness)  # constraint sum |skewness_i| <= 1
+
+            with numpyro.plate('obs_plate'):
+                sine = SineBivariateVonMises(phi_loc=phi_loc, psi_loc=psi_loc,
+                                             phi_concentration=70 * phi_conc,
+                                             psi_concentration=70 * psi_conc,
+                                             weighted_correlation=corr_scale)
+                return numpyro.sample('phi_psi', SineSkewed(sine, skewness), obs=obs)
+
+    To ensure the skewing does not alter the normalization constant of the (sine bivariate von Mises) base
+    distribution the skewness parameters are constraint. The constraint requires the sum of the absolute values of
+    skewness to be less than or equal to one. We can use the :class:`~numpyro.distriubtions.transforms.L1BallTransform`
+    to achieve this.
+
+    In the context of :class:`~pyro.infer.SVI`, this distribution can freely be used as a likelihood, but use as
+    latent variables it will lead to slow inference for 2 and higher dim toruses. This is because the base_dist
+    cannot be reparameterized.
+
+    .. note:: An event in the base distribution must be on a d-torus, so the event_shape must be `(d,)`.
+
+    .. note:: For the skewness parameter, it must hold that the sum of the absolute value of its weights for an event
+        must be less than or equal to one. See eq. 2.1 in [1].
+
+    ** References: **
+        1. Sine-skewed toroidal distributions and their application in protein bioinformatics
+            Ameijeiras-Alonso, J., Ley, C. (2019)
+
+    :param numpyro.distributions.Distribution base_dist: base density on a d-dimensional torus. Supported base
+        distributions include: 1D :class:`~numpyro.distributions.VonMises`,
+        :class:`~numnumpyro.distributions.SineBivariateVonMises`, 1D :class:`~numpyro.distributions.ProjectedNormal`,
+        and :class:`~numpyro.distributions.Uniform` (-pi, pi).
+    :param jax.numpy.array skewness: skewness of the distribution.
+    """
+
+    arg_constraints = {
+        "skewness": constraints.independent(constraints.interval(-1.0, 1.0), 1)
```

Comment:
I think we can use `constraints.l1_ball` here.

Reply:
That makes sense; I'll update it.